### PR TITLE
sql: two mechanical refactorings of sql errors and sql cluster settings

### PR DESCRIPTION
**Pull request by Xiang-Gu**
This PR contains two commits that are meant to be preparation work for supporting `CREATE DATABASE` in declarative schema changer. They are less controversial so I'd like to merge them in first to make `CREATE DATABASE` PR leaner. The idea is to move them to a different package so that `CREATE DATABASE` in declarative schema changer can import them (instead of importing whatever "hefty" package they currently are).
See individual commits for details.

Informs #103607

### CODE REVIEW

**Review by annrpom — APPROVED** [review]
this change lgtm

File: pkg/sql/sqlclustersettings/clustersettings.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package sqlclustersettings
+
+import "github.com/cockroachdb/cockroach/pkg/settings"
+
+// DefaultPrimaryRegionClusterSettingName is the name of the cluster setting that returns
```
[the remark this answers is not in the document]
Comment (Xiang-Gu) [reply pkg/sql/sqlclustersettings/clustersettings.go]: done

**Comment by Xiang-Gu** [comment]
TFTR!

bors r+

**Pull request merged**